sources/packages/api/src/services/application/application.service.ts

```diff
@@ -356,12 +360,26 @@ export class ApplicationService extends RecordDataModelService<Application> {
       })
       .getOne();
   }
-
+  /**
+   * Fetches application by applicationId and userId|studentId.
+   * When studentId is supplied, studentId is preferred as search criteria.
+   * When studentId is not supplied, then userId is used.
+   * @param applicationId
+   * @param userId
+   * @param studentId
+   * @returns
+   */
   async getApplicationByIdAndUser(
     applicationId: number,
-    userId: number,
+    userId?: number,
+    studentId?: number,
   ): Promise<Application> {
-    const application = await this.repo
+    if (!userId && !studentId) {
+      throw new UnprocessableEntityException(
```

Comment:
We are usually not returning HTTP exceptions from services.
The validation is really valid but I think that a regular Error exception would be better.
Not providing at least one would be more likely a developer mistake while consuming the method.

Reply:
I noticed that a CustomNamedError is created.
A simple Error would be ok IMO. I am resolving the topic either way.